src/sentry/profiles/task.py

```diff
@@ -346,6 +365,46 @@ def _process_symbolicator_results_for_rust(profile: Profile, stacktraces: List[A
             original["frames"] = symbolicated["frames"]
 
 
+"""
+This function returns a map {index: [indexes]} that will let us replace a specific
+frame index with (potentially) a list of frames indices that originated from that frame.
+
+The reason for this is that the frame from the SDK exists "physically",
+and symbolicator then synthesizes other frames for calls that have been inlined
+into the physical frame.
+
+Example:
+
+`
+fn a() {
+b()
+}
+fb b() {
+fn c_inlined() {}
+c_inlined()
+}
+`
+
+this would yield the following from the SDK:
+b -> a
+
+after symbolication you would have:
+c_inlined -> b -> a
+
+The sorting order is callee to caller (child to parent)
+"""
+
+
+def get_frame_index_map(frames: List[dict[str, Any]]) -> dict[int, List[int]]:
+    index_map: dict[int, List[int]] = {}
+    for i, frame in enumerate(frames):
+        original_idx = frame["original_index"]
+        idx_list = index_map.get(original_idx, [])
+        idx_list.append(i)
+        index_map[original_idx] = idx_list
```

Comment:
Is the order of this `idx_list` guaranteed to be correct? Could the inlined frame end up before the outer frame somehow?

Reply:
yes, it's guaranteed. I've double checked that with @Swatinem

Reply:
If we assume the ordering is from child to parent, as the comment above says then yes.

But this is a good idea to double-check, I believe the stack trace direction / order can be confusing sometimes.

Reply:
That might be something we have to double check since the order of the frames is now determined by the SDK where we have had a requirement on the ordering.